### PR TITLE
[Bug](predicate) add double predicate creator

### LINKED ISSUE #15761: [Bug] Be coredump when countering order by using double column
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

commit id:93b941baeb6afd00a1c005c12a0ef9d3ee89deb6

### What's Wrong?

Run the following query would crash BE.
```sql
SELECT ref_0.`supplycost`
FROM regression_test_query_p0_limit.tpch_tiny_partsupp AS ref_0
ORDER BY ref_0.`supplycost`
LIMIT 10;
```

### What You Expected?

Return results.

### How to Reproduce?

Run the query mentioned in What's Wrong

### Anything Else?

```
 1# os::Linux::chained_handler(int, siginfo*, void*) in /mnt/disk1/yuejing/downloads/jdk1.8.0_131/jre/lib/amd64/server/libjvm.so
 2# JVM_handle_linux_signal in /mnt/disk1/yuejing/downloads/jdk1.8.0_131/jre/lib/amd64/server/libjvm.so
 3# signalHandler(int, siginfo*, void*) in /mnt/disk1/yuejing/downloads/jdk1.8.0_131/jre/lib/amd64/server/libjvm.so
 4# 0x00007F59D9A95400 in /lib64/libc.so.6
 5# doris::ColumnPredicate* doris::create_comparison_predicate<(doris::PredicateType)4>(doris::TabletColumn const&, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, doris::MemPool*) at /mnt/disk1/yuejing/projects/doris/be/src/olap/predicate_creator.h:240
 6# doris::parse_to_predicate(std::shared_ptr<doris::TabletSchema>, doris::TCondition const&, doris::MemPool*, bool) in /mnt/disk1/yuejing/projects/doris/output/be/lib/doris_be
 7# doris::vectorized::RuntimePredicate::update(doris::vectorized::Field const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool) at /mnt/disk1/yuejing/projects/doris/be/src/runtime/runtime_predicate.cpp:146
 8# doris::vectorized::VSortNode::sink(doris::RuntimeState*, doris::vectorized::Block*, bool) in /mnt/disk1/yuejing/projects/doris/output/be/lib/doris_be
 9# doris::vectorized::VSortNode::open(doris::RuntimeState*) at /mnt/disk1/yuejing/projects/doris/be/src/vec/exec/vsort_node.cpp:166
10# doris::PlanFragmentExecutor::open_vectorized_internal() at /mnt/disk1/yuejing/projects/doris/be/src/runtime/plan_fragment_executor.cpp:274
11# doris::PlanFragmentExecutor::open() at /mnt/disk1/yuejing/projects/doris/be/src/runtime/plan_fragment_executor.cpp:246
12# doris::FragmentExecState::execute() at /mnt/disk1/yuejing/projects/doris/be/src/runtime/fragment_mgr.cpp:250
13# doris::FragmentMgr::_exec_actual(std::shared_ptr<doris::FragmentExecState>, std::function<void (doris::RuntimeState*, doris::Status*)>) at /mnt/disk1/yuejing/projects/doris/be/src/runtime/fragment_mgr.cpp:490
14# std::_Function_handler<void (), doris::FragmentMgr::exec_plan_fragment(doris::TExecPlanFragmentParams const&, std::function<void (doris::RuntimeState*, doris::Status*)>)::$_3>::_M_invoke(std::_Any_data const&) at /mnt/disk1/yuejing/projects/ldb_toolchain/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/std_function.h:291
15# doris::ThreadPool::dispatch_thread() in /mnt/disk1/yuejing/projects/doris/output/be/lib/doris_be
16# doris::Thread::supervise_thread(void*) at /mnt/disk1/yuejing/projects/doris/be/src/util/thread.cpp:454
17# start_thread in /lib64/libpthread.so.0
18# __GI___clone in /lib64/libc.so.6
```

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ByteYue**
# Proposed changes

Issue Number: close #15761 
Add one double predicator the same as integer predicate creator.
## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/predicate_creator.h [hunk 1]
```diff
@@ -163,6 +166,9 @@ inline std::unique_ptr<PredicateCreator<ConditionType>> get_creator(const FieldT
     case OLAP_FIELD_TYPE_LARGEINT: {
         return std::make_unique<IntegerPredicateCreator<TYPE_LARGEINT, PT, ConditionType>>();
     }
+    case OLAP_FIELD_TYPE_DOUBLE: {
```
Comment (BiteTheDDDDt) [inline be/src/olap/predicate_creator.h:169]: Maybe need add FLOAT also

Comment (ByteYue) [reply be/src/olap/predicate_creator.h:172]: > Maybe need add FLOAT also

😂My bad, I added both float and double in another machine but only added double in the machine where i created this pr.
Already added.

Comment (BiteTheDDDDt) [reply be/src/olap/predicate_creator.h:172]: > 

Better remove DoulbePredicatrCreator and use IntegerPredicateCreator directly on `DOUBLE`.
btw `DoulbePredicatrCreator` seems have a typo. XD

Comment (ByteYue) [reply be/src/olap/predicate_creator.h:169]: > 

> Better remove DoulbePredicatrCreator and use IntegerPredicateCreator directly on `DOUBLE`. 

OK. Already changed.
>btw `DoulbePredicatrCreator` seems have a typo. XD

😂Oh... My bad. I didn't notice it.

File: be/src/olap/predicate_creator.h [hunk 2]
```diff
@@ -64,7 +64,14 @@ class IntegerPredicateCreator : public PredicateCreator<ConditionType> {
 private:
     static CppType convert(const std::string& condition) {
         CppType value = 0;
-        std::from_chars(condition.data(), condition.data() + condition.size(), value);
+        // because std::from_chars can't compile on macOS
+        if constexpr (std::is_same_v<CppType, double>) {
+            value = std::stod(condition, nullptr);
+        } else if constexpr (std::is_same_v<CppType, float>) {
+            value = std::stof(condition, nullptr);
```
Comment (ByteYue) [inline be/src/olap/predicate_creator.h:71]: It's curious that std::from_char can't not pass the compilation on MacOS so i have to use specialization for double and float

**Review by BiteTheDDDDt — APPROVED** [review]
LGTM

**Pull request merged**